File: dmn-core/src/test/resources/tck/cl3/expected/java/dmn/0017-tabletests/PriceInRangeRuleOutput.java

import java.util.*;

@javax.annotation.Generated(value = {"decisionTableRuleOutput.ftl", "priceInRange"})
public class PriceInRangeRuleOutput extends com.gs.dmn.runtime.RuleOutput {
    private String priceInRange;
    private Integer priceInRangePriority;

    public PriceInRangeRuleOutput(boolean matched) {
        super(matched);
    }

    public String getPriceInRange() {
        return this.priceInRange;
    }
    public void setPriceInRange(String priceInRange) {
        this.priceInRange = priceInRange;
    }

    public Integer getPriceInRangePriority() {
        return this.priceInRangePriority;
    }
    public void setPriceInRangePriority(Integer priceInRangePriority) {
        this.priceInRangePriority = priceInRangePriority;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        PriceInRangeRuleOutput other = (PriceInRangeRuleOutput) o;
        if (this.getPriceInRange() != null ? !this.getPriceInRange().equals(other.getPriceInRange()) : other.getPriceInRange() != null) return false;
        if (this.getPriceInRangePriority() != null ? !this.getPriceInRangePriority().equals(other.getPriceInRangePriority()) : other.getPriceInRangePriority() != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = 0;
        result = 31 * result + (this.getPriceInRange() != null ? this.getPriceInRange().hashCode() : 0);
        result = 31 * result + (this.getPriceInRangePriority() != null ? this.getPriceInRangePriority().hashCode() : 0);

        return result;
    }

    public String toString() {
        StringBuilder result_ = new StringBuilder("(matched=" + isMatched());
        result_.append(String.format(", priceInRange='%s'", priceInRange));
        result_.append(")");
        return result_.toString();
    }

    @Override
    public List<com.gs.dmn.runtime.RuleOutput> sort(List<com.gs.dmn.runtime.RuleOutput> matchedResults_) {
        List<com.gs.dmn.runtime.Pair<String, Integer>> priceInRangePairs = new ArrayList<>();
        matchedResults_.forEach(matchedResult_ -> {
            priceInRangePairs.add(new com.gs.dmn.runtime.Pair(((PriceInRangeRuleOutput)matchedResult_).getPriceInRange(), ((PriceInRangeRuleOutput)matchedResult_).getPriceInRangePriority()));
        });
        priceInRangePairs.sort(new com.gs.dmn.runtime.PairComparator());

        List<com.gs.dmn.runtime.RuleOutput> result_ = new ArrayList<com.gs.dmn.runtime.RuleOutput>();
        for(int i=0; i<matchedResults_.size(); i++) {
            PriceInRangeRuleOutput output_ = new PriceInRangeRuleOutput(true);
            output_.setPriceInRange(priceInRangePairs.get(i).getLeft());
            output_.setPriceInRangePriority(priceInRangePairs.get(i).getRight());
            result_.add(output_);
        }
        return result_;
    }
}

File: dmn-core/src/test/resources/tck/cl3/expected/kotlin/dmn/0017-tabletests/PriceInRangeRuleOutput.kt

@javax.annotation.Generated(value = ["decisionTableRuleOutput.ftl", "priceInRange"])
class PriceInRangeRuleOutput(matched: Boolean) : com.gs.dmn.runtime.RuleOutput(matched) {
    @com.fasterxml.jackson.annotation.JsonProperty("priceInRange")
    var priceInRange: String? = null
    var priceInRangePriority: Int? = 0

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as PriceInRangeRuleOutput
        if (if (this.priceInRange != null) this.priceInRange != other.priceInRange else other.priceInRange != null) return false
        if (if (this.priceInRangePriority != null) this.priceInRangePriority != other.priceInRangePriority else other.priceInRangePriority != null) return false

        return true
    }

    override fun hashCode(): Int {
        var result = 0
        result = 31 * result + (if (this.priceInRange != null) this.priceInRange.hashCode() else 0)
        result = 31 * result + (if (this.priceInRangePriority != null) this.priceInRangePriority.hashCode() else 0)

        return result
    }

    override fun toString(): String {
        val result_ = StringBuilder("(matched=" + isMatched())
        result_.append(String.format(", priceInRange='%s'", priceInRange))
        result_.append(")")
        return result_.toString()
    }

    override fun sort(matchedResults_: MutableList<com.gs.dmn.runtime.RuleOutput>): MutableList<com.gs.dmn.runtime.RuleOutput> {
        val priceInRangePairs: MutableList<com.gs.dmn.runtime.Pair<String?, Int?>> = ArrayList()
        matchedResults_.forEach({ (it as PriceInRangeRuleOutput)
            priceInRangePairs.add(com.gs.dmn.runtime.Pair(it.priceInRange, it.priceInRangePriority))
        })
        priceInRangePairs.sortWith(com.gs.dmn.runtime.PairComparator())

        val result_: MutableList<com.gs.dmn.runtime.RuleOutput> = ArrayList<com.gs.dmn.runtime.RuleOutput>()
        for(i in 0 until matchedResults_.size) {
            var output_ = PriceInRangeRuleOutput(true)
            output_.priceInRange = priceInRangePairs.get(i).getLeft()
            output_.priceInRangePriority = priceInRangePairs.get(i).getRight()
            result_.add(output_)
        }
        return result_
    }
}
